src/App.tsx

```diff
@@ -1,10 +1,82 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  FormEvent,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import cn from 'classnames';
+
 import { UserWarning } from './UserWarning';
+import { FilterBy, Todo } from './types/Todo';
+import { getTodos } from './api/todos';
+import { wait } from './utils/fetchClient';
+import { filterTodos } from './services/todos';
 
-const USER_ID = 0;
+const USER_ID = 11144;
 
 export const App: React.FC = () => {
+  const [errorMessage, setErrorMessage] = useState('');
+  const [loading] = useState(true);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState<FilterBy>(FilterBy.ALL);
+  const [isAllCompleted, setIsAllCompleted] = useState(false);
+  const [isSomeCompleted, setIsSomeCompleted] = useState(false);
+  const [isNotificationOpen, setIsNotificationOpen] = useState(false);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  let preparedTodos = filterTodos(todos, filterBy);
+
+  function showNotificationBlock() {
+    setIsNotificationOpen(true);
+
+    wait(3000)
+      .then(() => {
+        setIsNotificationOpen(false);
+        setErrorMessage('');
+      });
+  }
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage('Can not load data');
+        showNotificationBlock();
+      });
+  }, []);
+
+  useEffect(() => {
+    preparedTodos = filterTodos(todos, filterBy);
+  }, [todos, filterBy]);
```

Comment:
you don't really need this hook because
`let preparedTodos = filterTodos(todos, filterBy);`
this line will set `preparedTodos` whenever this component is being rendered again